unit-testing/src/mock-function/lib/insert.spec.ts

```diff
@@ -0,0 +1,108 @@
+// insertDocument.test.ts
+import { Container } from '../data/connect-to-cosmos';
+import { createTestInputAndResult } from '../data/fake-data';
+import {
+  DbDocument,
+  DbError,
+  isDbDocument,
+  isDbError,
+  isVerificationErrors,
+  RawInput,
+} from '../data/model-schema';
+import { inputVerified } from '../data/verify';
+import { insertDocument } from './insert';
+
+// Mock the dependencies
+jest.mock('../data/connect-to-cosmos', () => ({
+  connectToContainer: jest.fn(),
+  getUniqueId: jest.fn().mockReturnValue('unique-id'),
+  Container: jest.fn(),
+}));
+
+jest.mock('../data/verify', () => ({
+  inputVerified: jest.fn(),
+}));
+
+describe('insertDocument', () => {
+  let mockContainer: jest.Mocked<Container>;
+
+  beforeEach(() => {
+    // Clear all mocks before each test
+    jest.clearAllMocks();
+
+    mockContainer = {
+      items: {
+        create: jest.fn(),
+      },
+    } as unknown as jest.Mocked<Container>;
+  });
+
+  it('should return verification error if input is not verified', async () => {
+    (inputVerified as jest.Mock).mockReturnValue(false);
+
+    // wrong shape of doc
+    const doc = { name: 'test' };
+
+    const returnedFunctionResult = await insertDocument(
```

Comment:
```suggestion
    const insertDocumentResult = await insertDocument(
```

You'll need to change the names below but I think that's a little clearer